.buildkite/scripts/build-docs.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+# Configure the git author and committer information
+export GIT_AUTHOR_NAME='Buildkite CI'
+export GIT_AUTHOR_EMAIL='buildkite@elasticsearch-ci.elastic.co'
+export GIT_COMMITTER_NAME=$GIT_AUTHOR_NAME
+export GIT_COMMITTER_EMAIL=$GIT_AUTHOR_EMAIL
+
+
+build_args=""
+rebuild_opt=""
+skiplinkcheck_opt=""
+warnlinkcheck_opt=""
+
+# From https://github.com/elastic/infra/blob/master/ci/jjb/elasticsearch-ci/defs/elastic-docs/pull-requests.yml#L110
+# Per https://github.com/elastic/docs/issues/1821, always rebuild all
+# books for PRs to the docs repo, for now.
+# When https://github.com/elastic/docs/issues/1823 is fixed, this
+# should be removed and the original behavior restored.
+#rebuild_opt="--rebuild"
+
+if [[ "${REBUILD}" == 'true' ]]; then
+  rebuild_opt="--rebuild"
+fi
+
+if [[ "${SKIP_LINK_CHECK}" == 'true' ]]; then
+  skiplinkcheck_opt="--skiplinkcheck"
+fi
+
+if [[ "${ALLOW_BROKEN_LINKS}" == 'true' ]]; then
+  warnlinkcheck_opt="--warnlinkcheck"
+fi
+
+# When running on a branch or on main
+if [[ "${BUILDKITE_PULL_REQUEST}" == "false" ]]; then
+  # temporary pushing to staging instead of master until the migration is over
+  build_args+=" --target_branch staging"
+else
+  build_args+=" --target_branch docs_bk_${BUILDKITE_PULL_REQUEST}"
+  build_args+=" --announce_preview https://docs_bk_${BUILDKITE_PULL_REQUEST}.docs-preview.app.elstc.co/diff"
+  rebuild_opt=" --rebuild"
```

Comment:
I'm pretty sure don't want to do full rebuilds on PRs by default (though giving folks the option is probably a good thing). 

```suggestion
```